nameko/testing/pytest.py

```diff
@@ -84,30 +85,51 @@ def rabbit_manager(request):
 
 @pytest.yield_fixture()
 def rabbit_config(request, rabbit_manager):
+    import random
+    import string
     from kombu import pools
-    from nameko.testing.utils import (
-        reset_rabbit_vhost, reset_rabbit_connections,
-        get_rabbit_connections, get_rabbit_config)
+    from six.moves.urllib.parse import urlparse  # pylint: disable=E0611
+    from nameko.testing.utils import get_rabbit_connections
 
     amqp_uri = request.config.getoption('AMQP_URI')
 
-    conf = get_rabbit_config(amqp_uri)
+    uri = urlparse(amqp_uri)
+    username = uri.username
+    vhost = uri.path[1:]
 
-    reset_rabbit_connections(conf['vhost'], rabbit_manager)
-    reset_rabbit_vhost(conf['vhost'], conf['username'], rabbit_manager)
+    use_random_vost = (vhost == ":random:")
+
+    if use_random_vost:
+        vhost = "test_{}".format(
+            "".join(random.choice(string.ascii_lowercase) for _ in range(10))
+        )
+        amqp_uri = "{}://{}/{}".format(uri.scheme, uri.netloc, vhost)
+        rabbit_manager.create_vhost(vhost)
+        rabbit_manager.set_vhost_permissions(vhost, username, '.*', '.*', '.*')
+
+    conf = {
+        'AMQP_URI': amqp_uri,
+        'username': username,
+        'vhost': vhost
+    }
 
     yield conf
 
     pools.reset()  # close connections in pools
 
     # raise a runtime error if the test leaves any connections lying around
-    connections = get_rabbit_connections(conf['vhost'], rabbit_manager)
-    open_connections = [
-        conn for conn in connections if conn['state'] != "closed"
-    ]
-    if open_connections:
-        count = len(open_connections)
-        raise RuntimeError("{} rabbit connection(s) left open.".format(count))
+    try:
```

Comment:
this is a weird pattern... it's not clear if any of your instructions raise exceptions or if it's just your own `RuntimeError` that you're handling.

Anything that doesn't raise an Exception should be taken out of the `try` block.

Maybe you could remove the `vhosts` explicitly based on the count rather than use the `finally` clause from the `Exception` that you raise yourself? And don't you have to remove the `vhost` if it's a random one or not?

Also, would be useful to know which of the connections were left open in your exception msg.

is there an option to check for `conn['state'] == "open"` because this is ambiguous if there are other states.

does the `rabbit_manager` object not have an API itself to return connections, as it has one for `delete`? Else could you move your helper function logic to the `rabbit_manager`?

`use_random_vhost` appears to be handling a problem rather than solving it.

Why a `RuntimeError` rather than something more specific?